core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -234,18 +235,94 @@ sealed trait BitcoinSignerSingle[-SpendingInfo <: BitcoinUTXOSpendingInfoSingle]
     )
 
     signatureF.map { sig =>
-      (spendingInfoToSatisfy.signer.publicKey, sig)
+      PartialSignature(spendingInfoToSatisfy.signer.publicKey, sig)
     }
   }
 }
 
 object BitcoinSignerSingle {
 
+  /**
+    * Signs the PSBT's input at the given input with the signer, then adds it to the PSBT
+    * in a PartialSignature record
+    * @param psbt The PSBT to sign
+    * @param inputIndex Index of input to sign
+    * @param signer Function or private key used to sign the PSBT
+    * @param isDummySignature Do not sign the tx for real, just use a dummy signature, this is useful for fee estimation
+    * @param conditionalPath Represents the spending branch being taken in a ScriptPubKey's execution
+    * @return
+    */
+  def sign(
+      psbt: PSBT,
+      inputIndex: Int,
+      signer: Sign,
+      conditionalPath: ConditionalPath = ConditionalPath.NoConditionsLeft,
+      isDummySignature: Boolean = false)(
+      implicit ec: ExecutionContext): Future[PSBT] = {
+    // if already signed by this signer
+    if (psbt
+          .inputMaps(inputIndex)
+          .partialSignatures
+          .exists(_.pubKey == signer.publicKey)) {
+      throw new IllegalArgumentException(
```

Comment:
`Future.failed(new IllegalArgumentException)` and add the public key to the error message